dissect/target/plugins/apps/av/trendmicro.py

```diff
@@ -0,0 +1,122 @@
+
+    def __init__(self, target: Target) -> None:
+        super().__init__(target)
+
+    def check_compatible(self) -> bool:
+        if not self.target.fs.path(self.LOG_FOLDER).exists():
+            raise UnsupportedPluginError
+
+    @export(record=TrendMicroWFLogRecord)
+    def wflogs(self) -> Iterator[TrendMicroWFLogRecord]:
+        """Return Trend Micro Worry-free log history records.
+
+        Yields TrendMicroWFLogRecord with the following fields:
+            hostname (string): The target hostname.
+            domain (string): The target domain.
+            ts (datetime): timestamp.
+            threat (string): Description of the detected threat (if available).
+            message (string): Message as reported in the user interface
+            keywords (string): Unparsed fields that might be visible in user interface.
+            fkey (string): Foreign key for reference for further investigation.
+        """
+        with self.target.fs.path(self.LOG_FILE_INFECTIONS).open("rt") as f:
+            for lineno, line in enumerate(f.readlines()):
+                cells = line.split("<;>")
+                yield TrendMicroWFLogRecord(
+                    ts=from_unix(int(cells[9])),
+                    message=",".join(cells),
+                    threat=cells[2],
+                    keywords=",".join([cells[0], cells[2], cells[6], cells[7], cells[8]]),
+                    fkey=lineno,
+                )
+
+    @export(record=TrendMicroWFFirewallRecord)
+    def wffirewall(self) -> Iterator[TrendMicroWFFirewallRecord]:
+        """Return Trend Micro Worry-free firewall log history records.
+
+        Yields TrendMicroWFFirewallRecord with the following fields:
+            hostname (string): The target hostname.
+            domain (string): The target domain.
+            ts (datetime): timestamp.
+            local_ip (net.ipadress): Local IPv4/IPv6.
+            remote_ip (net.ipaddress): Remote IPv4/IPv6.
+            port (uint16): Port of suspicious connection.
+            direction (string): Direction of the traffic
+            path (string): Path to object that initiated/received connection
+            description (string): Description of the detected threat
+        """
+        for firewall_log in self.target.fs.glob_ext(self.LOG_FILE_FIREWALL):
+            with firewall_log.open() as log:
+                try:
+                    while entry := c_pfwlog.firewall_entry(log):
+                        yield TrendMicroWFFirewallRecord(
+                            ts=from_unix(entry.timestamp),
+                            local_ip=entry.local_ip.strip(b"\x00").decode("utf-8"),
+                            remote_ip=entry.remote_ip.strip(b"\x00").decode("utf-8"),
+                            port=entry.port,
+                            direction=("out" if entry.direction == b"\x01" else "in"),
+                            path=entry.path.strip(b"\x00").decode(),
```

Comment:
```suggestion
                            path=path.from_windows(entry.path.strip(b"\x00").decode()),
```

The result you are currently getting in your testcase is not what you think it is ;-) It is a path where the `\`'s are part of the filename and not path separators.

Also don't forget to add a 
```
from flow.record.fieldtypes import path
```
in the imports.

Would be interesting to see how filenames with e.g. Chinese characters would be handled. Will it use the 8+3 file name "encoding" also in that instance?

Reply:
Worse, it will just store ??? (3F) internally. Only supports cp1252.